src/python/pants/base/specs.py

```diff
@@ -201,14 +201,14 @@ def make_glob_patterns(self, address_mapper):
 }
 
 
-def more_specific(spec1: Spec, spec2: Spec) -> Spec:
+def more_specific(spec1: Optional[Spec], spec2: Spec) -> Spec:
```

Comment:
`spec2` should also be `Optional` I believe, according to the comment on line 210 that either may be None.

Reply:
However, this will cause MyPy to flag an edge case for us: what happens if both specs are `None`? We could introduce a runtime check to `raise ValueError`, which would IMO be the right thing to do. Otherwise, use `cast` to get MyPy to ignore it.

Reply:
That's correct, but `spec2` is never `None` in the only place that we use this method, so I thought that enforcing that via a type check and not introducing a failure case would make more sense.